### PR TITLE
use pickle of ag to transport information

### PR DESCRIPTION
**Pull request by kain88-de**
Fixes #

Changes made in this Pull Request:
 - 

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

File: pmda/parallel.py [hunk 1]
```diff
@@ -343,11 +344,11 @@ def run(self,
             np.array([el[3] for el in res]), time_prepare, conclude.elapsed)
         return self
 
-    def _dask_helper(self, start, stop, step, indices, top, traj):
+    def _dask_helper(self, start, stop, step, pickles, top, traj):
         """helper function to actually setup dask graph"""
         with timeit() as b_universe:
             u = mda.Universe(top, traj)
-            agroups = [u.atoms[idx] for idx in indices]
+            agroups = [pickle.loads(idx) for idx in pickles]
```
[the remark this answers is not in the document]
Comment (richardjgowers) [reply pmda/parallel.py]: @kain88-de it's not actually hashing the Universe, it's just creating a uuid when the Universe is created.  So when you create a new Universe from the same files it "hashes" different.

[the remark this answers is not in the document]
Comment (kain88-de) [reply pmda/parallel.py]: So I also need to pickle the universe?

**Comment by richardjgowers** [comment]
or use the anchor thingy, it lets you define the hash

**Comment by kain88-de** [comment]
I have no idea how to use the *anchor thingy*.  The documentation isn't clear on this.

**Comment by kain88-de** [comment]
So I cannot pickle a Universe object because `__setstate__` and `__getstate__` functions are missing. This leaves the *anchor thingy*. If you can explain me how to use it I will give it a try.

**Comment by kain88-de** [comment]
@richardjgowers 

Anchor thingy works about 90% of the time to unpickle an atomgroup. Unfortunately, the results are really off now. Did I make a mistake in using the anchors? I guessed my usage from the code in MDAnalysis.

Also this looks like an error in MDAnalysis.

```python
 RuntimeError: Couldn't find a suitable Universe to unpickle AtomGroup onto with Universe hash 'c66d4379-9abc-46d6-91da-b31dc1a37af0'.  Available hashes: c66d4379-9abc-46d6-91da-b31dc1a37af0
```

**Comment by richardjgowers** [comment]
@kain88-de yeah that looks right.  The tests look like they're failing because the trajectory isn't being used

That second one looks like a bug, maybe it's a bytes/str difference?

**Comment by kain88-de** [comment]
what do you mean with trajectory not being used?

about bytes/strings I only use the MDAnalysis functions. But the python 2.7
test fail with the same error message. So maybe something more fundamental.

**Comment by richardjgowers** [comment]
The contacts is coming out as ones, which (I’m guessing) means the same
frame is being used for all frames, so the contact is constant

**Pull request closed without merging**